enforcer_interface.go

```diff
@@ -89,10 +89,14 @@ type IEnforcer interface {
 	HasPolicy(params ...interface{}) bool
 	HasNamedPolicy(ptype string, params ...interface{}) bool
 	AddPolicy(params ...interface{}) (bool, error)
+	AddPolicies(rules [][]string) ([]bool, []error)
```

Comment:
You missed 4 functions: AddGroupingPolicies, AddNamedGroupingPolicies, RemoveNamedGroupingPolicies, RemoveNamedGroupingPolicies

Reply:
Use the method of `Adapter` is enough.